src/main/scala/com/gu/fastly/Lambda.scala

```diff
@@ -23,6 +23,60 @@ class Lambda {
   private val cloudWatchClient = AmazonCloudWatchClientBuilder.defaultClient
   private val snsClient = AmazonSNSClientBuilder.defaultClient
 
+  private def raiseAllThePurges(event: Event): Boolean = {
+    val contentType = extractUpdateContentType(event)
+    val purgeType = if (event.eventType == EventType.Delete) {
+      Hard
+    } else {
+      Soft
+    }
+    val aliasPaths = extractAliasPaths(event)
+    val dotcomCanonicalPurge = sendFastlyPurgeRequest(event.payloadId, purgeType, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(event.payloadId), config.fastlyDotcomApiKey, contentType)
+    val jsonCanonicalPurge = sendFastlyPurgeRequestForAjaxFile(event.payloadId, contentType)
+    val mapiCanonicalPurge = sendFastlyPurgeRequest(event.payloadId, purgeType, config.fastlyMapiServiceId, makeMapiSurrogateKey(event.payloadId), config.fastlyMapiApiKey, contentType)
+    val aliasPurges = aliasPaths.fold(Seq.empty[Boolean])({
+      _.map { aliasPath =>
+        val dotcomAliasPurge = sendFastlyPurgeRequest(aliasPath, purgeType, config.fastlyDotcomServiceId, makeDotcomSurrogateKey(aliasPath), config.fastlyDotcomApiKey, contentType)
+        val jsonAliasPurge = sendFastlyPurgeRequestForAjaxFile(aliasPath, contentType)
+        val mapiAliasPurge = sendFastlyPurgeRequest(aliasPath, purgeType, config.fastlyMapiServiceId, makeMapiSurrogateKey(aliasPath), config.fastlyMapiApiKey, contentType)
+        dotcomAliasPurge && jsonAliasPurge && mapiAliasPurge
+      }
+    })
+    // all or nothing result
+    dotcomCanonicalPurge && jsonCanonicalPurge && mapiCanonicalPurge && !aliasPurges.contains(false)
+  }
+
+  private def makeContentDecachedEventsFromCrierEvent(crierEvent: com.gu.crier.model.event.v1.Event): Seq[ContentDecachedEvent] = {
+    // if an update or delete from Crier features a content item with
+    // aliasPaths, we must raise de-cache events for the current path and
+    // all aliases
+    val fastlyEventType = crierEvent.eventType match {
+      case EventType.Delete => com.gu.fastly.model.event.v1.EventType.Delete
+      case _ => com.gu.fastly.model.event.v1.EventType.Update
```

Comment:
Possibly an exhaustive match would be clearer for people reading the code?

Reply:
Not sure what this means but happy to talk a call to be educated.

Reply:
sorry, should have been more explicit - I'm presuming the the other event type you're matching on is `Update`? In which case you could just match on that instead of `_`.